src/main/io/osd.c

```diff
@@ -197,7 +197,8 @@ static const uint8_t osdElementDisplayOrder[] = {
     OSD_NUMERICAL_HEADING,
     OSD_NUMERICAL_VARIO,
     OSD_COMPASS_BAR,
-    OSD_ANTI_GRAVITY
+    OSD_ANTI_GRAVITY,
+    OSD_LOG_STATUS,
```

Comment:
Elements in this array are meant to be displayed unconditionally.  In other words they always display and are not dependent on other conditions.  That's not true with your new element as it depends on a number of conditions.  Remove the element from here and move your conditional logic/`#ifdef` from `osdDrawSingleElement()` to `osdDrawElements()`.  There's no reason to iterate through the element if for example the board doesn't have blackbox capability.